cmake/Tools/Platform/Android/android_post_build.py

```diff
@@ -0,0 +1,259 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+#
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+import argparse
+import re
+import shutil
+import sys
+
+import platform
+import logging
+
+from packaging.version import Version
+from pathlib import Path
+
+logging.basicConfig(format='%(levelname)s: %(message)s', level=logging.INFO)
+logger = logging.getLogger('o3de.android')
+
+ANDROID_ARCH = 'arm64-v8a'
+
+ASSET_MODE_PAK     = 'PAK'
+ASSET_MODE_LOOSE   = 'LOOSE'
+ASSET_PLATFORM_KEY = 'android'
+
+MINIMUM_ANDROID_GRADLE_PLUGIN_VER = Version("8.0")
+
+IS_PLATFORM_WINDOWS = platform.system() == 'Windows'
+
+PAK_FILE_INSTRUCTIONS = "Make sure to create release bundles (Pak files) before building and deploying to an android device. Refer to " \
+                        "https://www.docs.o3de.org/docs/user-guide/packaging/asset-bundler/bundle-assets-for-release/ for more" \
+                        "information."
+
+
+class AndroidPostBuildError(Exception):
+    pass
+
+
+def copy_or_create_link(src: Path, tgt: Path):
+    """
+    Copy or create a link/junction depending on the source type. If this is a file, then perform file copy from the
+    source to the target. If it is a directory, then create a junction(windows) or symlink(linux) from the source to
+    the target
+
+    :param src:             The source to copy/link from
+    :param tgt:             The target tp copy/link to
+    """
+
+    assert src.exists()
+    assert not tgt.exists()
+
+    full_src_path = str(src.resolve().absolute())
+    full_tgt_path = str(tgt.resolve().absolute())
+
+    try:
+        if src.is_file():
+            shutil.copy2(src=full_src_path,
+                         dst=full_tgt_path,
+                         follow_symlinks=True)
+            logger.info(f"Copied {full_src_path} to {full_tgt_path}")
```

Comment:
Technically instead of copying the files a hard link can be created for the source file and that speed up the asset layout.
You can use the [pathlib.Path.hardlink_to](https://docs.python.org/3/library/pathlib.html?highlight=python%20hardlink#pathlib.Path.hardlink_to) function to create a hardlink

Here is some information about hard links https://linuxhandbook.com/hard-link/

Reply:
Thats pretty cool python 3.10 hotness